src/components/withDelayToggleButtonState.js

```diff
@@ -0,0 +1,82 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+
+const withDelayToggleButtonStatePropTypes = {
+    /** A value whether the button state is complete */
+    isDelayButtonStateComplete: PropTypes.bool.isRequired,
+
+    /** A function to call to change the complete state */
+    toggleDelayButtonState: PropTypes.func.isRequired,
+};
+
+export default function (WrappedComponent) {
+    class WithDelayToggleButtonState extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                isDelayButtonStateComplete: false,
+            };
+            this.toggleDelayButtonState = this.toggleDelayButtonState.bind(this);
+        }
+
+        componentWillUnmount() {
+            if (!this.resetButtonStateCompleteTimer) {
+                return;
+            }
+
+            clearTimeout(this.resetButtonStateCompleteTimer);
+        }
+
+        /**
+         * @param {Boolean} [resetAfterDelay] Impose delay before toggling state
+         */
+        toggleDelayButtonState(resetAfterDelay) {
```

Comment:
~provide default value for optional param.~

Reply:
or rather why is the param doc optional. Looking at the code i don't see any flow where its optional so i think you can just remove `[resetAfterDelay]` from the JS doc.